pytorch_lightning/overrides/data_parallel.py

```diff
@@ -187,15 +189,26 @@ def __init__(self, pl_module: LightningModule):
         self.module = pl_module
 
     def forward(self, *inputs, **kwargs):
-        if self.module.training:
+
+        if self.module.running_stage == RunningStage.TRAINING:
```

Comment:
Not sure if this is cleaner, but can we not define properties for this within the LightningModule like so: 

```python
class LightningModule(nn.Module):
...
   @property
    def training():
        return self.running_stage == RunningStage.Training
```
will reduce duplicated lines I think

Reply:
Could do, but training is protected right ? I thought it could be confusing.
Also, RunningStage seems less ambiguous as it is a Trainer stage.

Reply:
all fair!